lib/engine/game/g_1822_mx/step/auction_ndem_tokens.rb

```diff
@@ -253,11 +255,17 @@ def process_merge(action)
             @available_ndem_tokens.delete(@token_up_for_bid)
             @token_up_for_bid = nil
             @player_step_order.append(@player_step_order.shift)
-            @remaining_choosers = @player_step_order.select { |p| player_can_purchase_any_token?(p) }
+            @remaining_choosers = @player_step_order.dup
           end
 
           def auto_actions(entity)
             return [Engine::Action::Pass.new(entity)] if @remaining_choosers.empty?
+
+            if !@token_up_for_bid
+              return [Engine::Action::Pass.new(entity)] unless player_can_purchase_any_token?(@remaining_choosers[0])
+            elsif !player_can_purchase_token?(@remaining_bidders[0], @token_up_for_bid) && !auction_over?
+              [Engine::Action::Pass.new(entity)]
+            end
```

Comment:
Can you clean this method up? These are all conditions where a pass is added as an auto action. It'd be easier to understand the code if there was a condition block with each of these possibilities.

Reply:
Cleaned up